app/Builders/Hierarchy/Hierarchy.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace App\Builders\Hierarchy;
+
+class Hierarchy
+{
+    private $id;
+    private $name;
+    private $model;
+    private $parentColumnName = "parent_id";
+
+    public function __construct(String $model)
+    {
+        $this->model = new $model();
+    }
+
+    public function SetParentColumn($parentColumnName)
+    {
+        $this->parentColumnName = $parentColumnName;
+        return $this;
+    }
+
+    public function SetIdNamePair(array $idNameArray)
```

Comment:
```suggestion
    public function SetIdNamePair(int $id, string $name)
```
agar bir nechta bo'sa locale ga qarab oladigan qilamz

Reply:
ozgartirildi